### PR TITLE
Add @ApiSecurity decorator for define SecurityDefinitions

### PR DESCRIPTION
**Pull request by isman-usoh**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #140
Pull Number: #142 #188 

## What is the new behavior?
- add `@ApiSecurity` Decorator
```ts
import { ApiSecurity } from "@nestjs/swagger"

@Controller('my')
@ApiSecurity({ type: 'basic', name: 'myBasic' })
@ApiSecurity({ type: 'apiKey', name: 'myBearer' })
@ApiSecurity({ type: 'oauth2', name: 'myOAuth2', scopes: ['read', 'write'] })
export class MyController {

    @Get('hello')
    @ApiSecurity({ type: 'basic', name: 'myBasic' })
    @ApiSecurity({ type: 'apiKey', name: 'myBearer' })
    @ApiSecurity({ type: 'oauth2', name: 'myOAuth2', scopes: ['read', 'write'] })
    async hello() {
        return "Hello"
    }
}
````

- add `addSecurity` method in `DocumentBuilder` for override security definitions
```ts
import { DocumentBuilder } from "@nestjs/swagger"

const options = new DocumentBuilder()
        .addSecurity('myAuth', {
            type: 'oauth2',
            flow: 'password',
            tokenUrl: 'http://localhost:9000/oauth2',
            scopes: {
              'read': 'Read scope',
              'write': 'Write scope',
            }
        })
     .build()
```

- Auto generate SecurityDefinitions
```json
{
   "swagger": "2.0",
   "securityDefinitions":{
      "myOAuth2":{
         "type":"oauth2",
         "description":"Auto generate SecurityDefinitions",
         "flow":"implicit",
         "authorizationUrl":"http://localhost:9000/oauth2",
         "scopes":{
            "read":"READ",
            "write":"WRITE"
         }
      },
      "myBearer":{
         "type":"apiKey",
         "description":"Auto generate SecurityDefinitions",
         "name":"access_token",
         "in":"query"
      },
      "myBasic":{
         "type":"basic",
         "description":"Auto generate SecurityDefinitions"
      }
   }
}
```

- compatible `@ApiBearerAuth` and `@ApiOAuth2Auth`
## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

**Comment by kamilmysliwiec** [comment]
Thank you! I'll review it soon

File: lib/decorators/api-bearer.decorator.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import { DECORATORS } from '../constants';
-import { createMixedDecorator } from './helpers';
+import { ApiSecurity } from './api-security.decorator';
 
+/**
+ * @deprecated Use @ApiSecurity
+ */
```
Comment (kamilmysliwiec) [inline lib/decorators/api-bearer.decorator.ts]: Do we really have to deprecate these decorators? (`@ApiBearerAuth()` and `@ApiOAuth2Auth()`)?

Comment (isman-usoh) [reply lib/decorators/api-bearer.decorator.ts]: I think it should be removed in the next version. for prevent confusion

**Comment by isman-usoh** [comment]
Install this PR
```
$ npm i --save @isman/nest-swagger
```

**Comment by kamilmysliwiec** [comment]
This feature has been added in 4.0.0 already. Hence, we can't merge your PR. Thanks for the contribution though!

**Pull request closed without merging**